sw/device/silicon_creator/lib/base/sec_mmio_unittest.cc

```diff
@@ -0,0 +1,189 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include "sw/device/silicon_creator/lib/base/sec_mmio.h"
+
+#include <array>
+#include <vector>
+
+#include "gtest/gtest.h"
+#include "sw/device/silicon_creator/lib/base/mock_abs_mmio.h"
+
+extern "C" {
+// This is an extern in the sec_mmio module.
+sec_mmio_ctx_t sec_mmio_ctx;
+
+// FIXME: Replace for shutdown Mock function.
+uint32_t fake_shutdown_count;
+}
+
+namespace sec_mmio_unittest {
+namespace {
+using ::testing::Each;
+using ::testing::Eq;
+using ::testing::Test;
+
+class SecMmioTest : public mask_rom_test::MaskRomTest {
+ protected:
+  void SetUp() override {
+    *shutdown_count_ = 0;
+    sec_mmio_init(+[] { ++fake_shutdown_count; });
+  }
+
+  uint32_t GetShutDownCount() { return *shutdown_count_; }
+
+  sec_mmio_ctx_t *ctx_ = &::sec_mmio_ctx;
+  uint32_t *shutdown_count_ = &::fake_shutdown_count;
+  mask_rom_test::MockAbsMmio mmio_;
+};
+
+TEST_F(SecMmioTest, Initialize) {
+  EXPECT_EQ(ctx_->check_count, 0);
```

Comment:
just wanted to point out that the global ctx is zero initialized. You may want to initialize it with different values (before init is called).

Reply:
fixed to initialize data with non-zero data for this test case.